app/web/i18n/namespaces.ts

```diff
@@ -0,0 +1,21 @@
+export const AUTH = "auth";
+export const COMMUNITIES = "communities";
+export const CONNECTIONS = "connections";
+export const DASHBOARD = "dashboard";
+export const DONATIONS = "donations";
+export const GLOBAL = "global";
+export const MESSAGES = "messages";
+export const PROFILE = "profile";
+export const SEARCH = "search";
+
+export const NAMESPACES = {
+  AUTH,
+  COMMUNITIES,
+  CONNECTIONS,
+  DASHBOARD,
+  DONATIONS,
+  GLOBAL,
+  MESSAGES,
+  PROFILE,
+  SEARCH,
+};
```

Comment:
```suggestion
export const NAMESPACES = [
  AUTH,
  COMMUNITIES,
  CONNECTIONS,
  DASHBOARD,
  DONATIONS,
  GLOBAL,
  MESSAGES,
  PROFILE,
  SEARCH,
];
```
In `next-18next.config.js`, it's expecting all namespaces in an array. Maybe we can do it here and re-use it in that config too so this becomes the source of truth?

Reply:
Or keep this as an object and have this be shared with `i18n/resources.ts`, and generate the namespace array from this in the config file, so they always stay in sync.

A bit of background of when I was made `i18n/resources.ts`: it was done as a convenience to provide custom typing for the `react-i18next-types.d.ts` file, so we get better autocomplete and thus better developer experience. I believe the `i18n/resources.ts` file pattern also helps to avoid incurring the penalty of having to import all of the English translation JSONs in the app, when not all of the JSONs are needed by the app at a particular point in time, as it's only used for TS typing and so should get tree shaked away in the final JS bundle.

Reply:
Nice. I think I prefer to have both the array and the object versions exported from namespaces. I'll update the PR, please let me know if that is ok.

Reply:
Ok, I removed the object and kept just the array for the next-i18next config.